pkg/reconciler/trigger/trigger.go

```diff
@@ -201,9 +203,14 @@ func (r *Reconciler) reconcile(ctx context.Context, t *v1alpha1.Trigger) error {
 		if err := trackAddressable(*t.Spec.Subscriber.Ref); err != nil {
```

Comment:
We might not need this trackAddressable anymore, since uriResolver does that. cc @n3wscott

Reply:
trackInNamespace use ensureTracking() to ensure that there is an informer watching and sending events to tracker for the concrete GVK. It also ensures that there is the corresponding lister for that informer. In the func URIFromDestination of uriResolver, it doesn't use any similar function like ensureTracking(). I'm not sure whether we can remove the  trackAddressable here.

Reply:
Is there anyway that I can do some manual test to see whether it works after we delete the trackAddressable? Should I do some changes for the resource of subscriber?

Reply:
You can subscribe to an object and have it change the resolved uri

Reply:
@n3wscott  I don't quite understand your meaning of "You can subscribe to an object and have it change the resolved uri". Could you give me some example?

Reply:
Do you mean I can do the manual test by letting Trigger subscribe to an object and then change the URI of the object?

Reply:
@grantr  @n3wscott  I did some manual tests after I deleted trackAddressable. I created Trigger which needs a subscriber `event-display`.At first it failed because the subscriber which the trigger needs is not present. Then I created the service `event-display`. The Trigger is able to reconcile again and becomes ready.

However, if i delete `event-display`, the status of Trigger won't change. It is still ready and Subscriber URI of status still exists.  I'm not sure whether this is the result we want. If it is a bug, the bug is not introduced by this PR. I tested the master branch. It's same. I checked the code, it is because we never change Subscriber URI when there is any change to the subscriber.

Reply:
I think that is not the result we want, but if the bug exists in master branch, we should create an issue to track it.